File: 75. Sort Colors/Solution.py
class Solution:
    # @param A a list of integers
    # @return nothing, sort in place
    def sortColors(self, A):
        if len(A) <= 1:
            return
            
        i = 0
        j = len(A) - 1
        zeroIndex = 0
        
        while i <= j:
            if A[i] == 0:
                self.swap(A, i, zeroIndex)
                i += 1
                zeroIndex += 1
            elif A[i] == 2:
                self.swap(A, i, j)
                j -= 1
            else:
                i += 1
                
        return
    
    def swap(self, A, i, j):
        temp = A[i]
        A[i] = A[j]
        A[j] = temp
            
        

File: 75. Sort Colors/Solution.cpp
class Solution {
public:
    void sortColors(int A[], int n) {
        if (n <= 1)
            return;
            
        int i = 0;
        int j = n - 1;
        int zeroIndex = 0;
        
        while (i <= j) {
            if (A[i] == 0) {
                swap(A, i, zeroIndex);
                i++;
                zeroIndex++;
            } else if (A[i] == 2) {
                swap(A, i, j);
                j--;
            } else {
                i++;
            }
        }
        return;
    }

private:
    void swap(int A[], int i, int j) {
        int temp = A[i];
        A[i] = A[j];
        A[j] = temp;
    }
};